src/test/java/core/basesyntax/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,109 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationService;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class RegistrationServiceImplTest {
+    private static final int LOGIN_MIN_LENGHT = 6;
+    private static final int USER_PWD_MIN_LENGHT = 6;
+    private static final int USER_MIN_AGE = 18;
+    private RegistrationService registrationService = new RegistrationServiceImpl();
+
+    private User user;
+
+    @BeforeEach
+    void setUp() {
+        user = new User();
+        user.setLogin("Alice66");
+        user.setAge(20);
+        user.setPassword("QWERT666");
+    }
+
+    @Test
+    void get_UserNull_notOk() {
+        user = null;
+        assertThrows(DataNotVaidExeption.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void getLogin_LessSixCharacters_notOk() {
+
+        user.setLogin("abcd");
+        assertThrows(DataNotVaidExeption.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void get_PwdLessSixCharacters_notOk() {
+
+        user.setPassword("qwert");
+        assertThrows(DataNotVaidExeption.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void get_UserAgeNotAdult_notOk() {
+
+        user.setAge(5);
+        assertThrows(DataNotVaidExeption.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void get_UserAgeLessZero_notOk() {
+        user.setAge(-1);
+        assertThrows(DataNotVaidExeption.class, () -> {
+            registrationService.register(user);
+        });
+
+    }
+
+    @Test
+    void get_UserLoginNotUnique_notOk() {
+        User user1 = new User();
+        user1.setLogin("Alice66");
+        user1.setAge(20);
+        user1.setPassword("QWERT666");
+        assertThrows(DataNotVaidExeption.class, () -> {
+            registrationService.register(user);
+            registrationService.register(user1);
+        });
+    }
```

Comment:
don't use the tested method twice, add directly via dao, as explained here
![image](https://github.com/mate-academy/jv-registration-validation/assets/49767049/09967cfe-d565-46bb-b579-aefe7170fac4)